### PR TITLE
Add check for validating dualstack configuration

### PR DESCRIPTION
**Pull request by rupeshsahuoc**

### CODE REVIEW

**Comment by snaiksat** [comment]
Thanks @rupeshsahuoc . Can you please update the commit message clearly indicating which combinations of single stack and dual stack inputs will be validated, which will pass the validation and which will fail the validation?

If multiple subnets are provided, we want don't want to incorrectly detect the intent that its dual stack (it could be multiple incorrect subnets for single stack, either v4 or v6) and then show them the message that they must provide *both* v4 and v6.

File: provision/acc_provision/acc_provision.py
Comment (PariyaAshok) [inline provision/acc_provision/acc_provision.py:2104]: We can get/fill this info once and use, no need call func has_contains_v6 multiple time, It would be better if we can have function which check IPV4 Only case, IPV6 Only case and dual stack case, for all other case we can just raise error with appropriate message.

**Review by PariyaAshok — CHANGES_REQUESTED** [review]

**Comment by snaiksat** [comment]
All ipv4 is not mentioned as a valid use case in the examples in the commit message. ;-)

File: provision/acc_provision/acc_provision.py
Comment (rupeshsahuoc) [reply provision/acc_provision/acc_provision.py]: addressed

File: provision/acc_provision/acc_provision.py
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:2230]: Although not part of this patch, there is a typo here, kubernetes version should be 1.21

File: provision/acc_provision/acc_provision.py
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:2074]: what if the address malformed, this conversion will throw an exception and acc-provision will exit with a stack trace?

File: provision/acc_provision/acc_provision.py
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:2235]: I am still not comfortable with this logic here. If the address is malformed, how would we know if the intent was to specify an ipv4 or ipv6 address? And if you cannot infer that conclusively, how can we infer that the intent was to create dual stack?

I would rather list out all the subnets we have processed indicating which ones we have recognized as ipv4, which we have recognized as ipv6, and if there is a mix of ipv4 and ipv6 subnets, indicate clearly which incorrect subnet(s) caused us to conclude that this is not a dual stack case.

**Review by snaiksat — CHANGES_REQUESTED** [review]

**Comment by snaiksat** [comment]
@rupeshsahuoc  please also add a UT to cover each of the valid and invalid scenarios you have listed in the commit message.

File: provision/acc_provision/acc_provision.py
Comment (rupeshsahuoc) [reply provision/acc_provision/acc_provision.py]: addressed

File: provision/acc_provision/acc_provision.py
Comment (rupeshsahuoc) [reply provision/acc_provision/acc_provision.py:2074]: addressed

File: provision/acc_provision/acc_provision.py
Comment (rupeshsahuoc) [reply provision/acc_provision/acc_provision.py]: addressed

File: provision/acc_provision/acc_provision.py
Comment (snaiksat) [reply provision/acc_provision/acc_provision.py:2074]: how? I still don't see a try/catch block...I see a check for empty string, but the ip address could be malformed for other, reasons, no?

File: provision/acc_provision/acc_provision.py
Comment (snaiksat) [reply provision/acc_provision/acc_provision.py]: how?

Comment (rupeshsahuoc) [reply provision/acc_provision/acc_provision.py]: This process involves verifying the correctness of all subnets. If any subnets are found to be malformed, an error will be thrown indicating the specific address that is malformed. Conversely, if all subnets are valid, the next step is to validate the dualstack configuration. In case any aspect of the configuration is incorrect, an error will be displayed highlighting the specific component that needs correction.

**Review by snaiksat — APPROVED** [review]

**Comment by snaiksat** [comment]
Thanks @rupeshsahuoc  for making the changes. @PariyaAshok currently merging is blocked, awaiting your review/approval. 

@rupeshsahuoc once merged please backport to the relevant branches.

**Review by PariyaAshok — APPROVED** [review]

**Pull request merged**
